test/models/template_test.rb

```diff
@@ -411,5 +413,42 @@ class TemplateTest < ActiveSupport::TestCase
         refute_includes @template.operatingsystem_ids, @os2.id
       end
     end
+
+    describe '::find_without_name_collision in subclasses' do
+      setup do
+        @org = FactoryBot.create(:organization, :name => 'TemplateOrg')
+        @empty = FactoryBot.create(:organization, :name => 'EmptyOrg')
+        @regular_template = FactoryBot.create(:provisioning_template, :name => 'regular template', :organizations => [@org])
+        @collision_template = FactoryBot.create(:provisioning_template, :name => 'collision template', :organizations => [@empty])
+        @common_template = FactoryBot.create(:provisioning_template, :name => 'common template', :organizations => [@empty, @org])
+        @before_org = Organization.current
+        Organization.current = @org
+      end
+
+      after do
+        Organization.current = @before_org
+      end
```

Comment:
Don't do this, we have `teardown :clear_current_taxonomies` on each test from ActiveSupport::TestCase already.